File: Library.Application/Borrowings/Dtos/BorrowingDetailDto.cs
namespace Library.Application.Borrowings.Dtos
{
	public record BorrowingDetailDto(
		int Id,
        BookBorrowingDto Book,
		string UserEmail,
		DateTime BorrowDate,
		DateTime DueDate,
		DateTime? ReturnDate,
		BorrowingStatus Status
	);
}

File: library-app-client/src/app/features/borrowing/dtos/borrowing-detail.dto.ts
import { BookBorrowingDto } from "../../book/dtos/book-borrowing.dto";
import { BorrowingStatus } from "../enums/borrowings-status.enum";

export interface BorrowingDetailDto {
  id: number;
  book: BookBorrowingDto;
  userEmail: string;
  borrowDate: string;
  dueDate: string;
  returnDate: string | null;
  status: BorrowingStatus;
}